astropy/time/core.py

```diff
@@ -109,6 +109,44 @@ class _LeapSecondsCheck(enum.Enum):
 _LEAP_SECONDS_LOCK = threading.RLock()
 
 
+def _compress_array_dims(arr):
+    """Compress array by allowing at most 2 * edgeitems + 1 in each dimension.
```

Comment:
It does seem we could have used `np.core.arrayprint._leading_trailing` if only we had support for `np.concatenate` -- maybe add a comment that this can be replaced by that once we do? See #8610...

Reply:
Done. I'll merge once this passes CI, thanks for the review.